### PR TITLE
chore: remove more code with TypeScript's `noUnusedLocals`

### PR DESCRIPTION
**Pull request by danielgent**
After using `knip` I thought I would set the typescript param `noUnusedLocals`

I don't like having these enabled permanently as IMHO it ruins my DX _but_ maybe we want these on our CI/CD, or at least now and again we do this

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -25,7 +25,8 @@
     ],
     "paths": {
       "*": ["node_modules/*"]
-    }
+    },
+    "noUnusedLocals": false
```
Comment (danielgent) [inline tsconfig.json:29]: :eyes: we could actually set this to true but I find it makes developing a living hell :smile:

File: scripts/import-users.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ const userSchema = z.object({
 
 type User = z.infer<typeof userSchema>;
 
-const tenantId = "qo0kCHUE8qAvpNPznuoRW";
+// const tenantId = "qo0kCHUE8qAvpNPznuoRW";
```
Comment (danielgent) [inline scripts/import-users.ts:43]: I figure on scripts I can comment them out

File: src/adapters/kysely/otps/list.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Database, OTP, SqlOTP } from "../../../types";
+import { Database, OTP } from "../../../types";
```
Comment (danielgent) [inline src/adapters/kysely/otps/list.ts:1]: this PR is mostly this kind of thing

File: src/token-grant-types/passwordGrant.ts
Comment (danielgent) [inline src/token-grant-types/passwordGrant.ts:1]: as mentioned above :wastebasket:

File: test/fixtures/oauth2Client.ts [hunk 4]
```diff
@@ -12,7 +12,9 @@ export function oAuth2ClientFactory(
 }
 
 class OAuth2ClientFixture implements IOAuth2Client {
+  //@ts-ignore
```
Comment (danielgent) [inline test/fixtures/oauth2Client.ts:15]: I figure for mocks we can tell typescript to ignore some rules

As we need to declare unused params in order to implement `IOAuth2Client`

File: test/integration/flows/code-flow.spec.ts [hunk 5]
```diff
@@ -131,15 +131,13 @@ describe("code-flow", () => {
     // now check silent auth works when logged in with code----------------------------------------
     const setCookiesHeader = tokenResponse.headers.get("set-cookie")!;
 
-    const {
-      accessToken: silentAuthAccessTokenPayload,
```
Comment (danielgent) [inline test/integration/flows/code-flow.spec.ts:135]: a param that I'm not using anywhere really :smile:

File: test/token-grant-types/passwordGrant.spec.ts
Comment (danielgent) [inline test/token-grant-types/passwordGrant.spec.ts:1]: as mentioned above

Comment (danielgent) [reply test/token-grant-types/passwordGrant.spec.ts:1]: a skipped test for a function that just throws an exception saying it's not done :smile: 

Like we're getting paid for every line of code :sunglasses:

**Review by markusahlstrand — APPROVED** [review]
Really nice :)

**Pull request merged**